### ISSUE TITLE
PostBuild failing randomly: "apphost.exe' as application host executable as it does not contain the expected placeholder byte sequence"

### ISSUE THREAD

**Issue by jwdonahue** [dotnet/runtime#49458]
I've been encountering this post build failure on about 33% of my builds, since taking some VS 2019 updates a few weeks back. See attached [BuildTest.zip](https://github.com/dotnet/sdk/files/6066601/BuildTest.zip) for 10 build logs where 3 of them failed. [build.zip](https://github.com/dotnet/sdk/files/6066619/build.zip) contains the batch file that drives the dotnet build/test. Any ideas what I can do to fix it? 

Example:

       PostBuild:
         copy d:\repo\SH-Core-Libs\CLI\Options\Example2\\Example2.ReadMe.txt d:\repo\SH-Core-Libs\\.BuildOutput\net5.0\Bin\Debug\
                 1 file(s) copied.
     6>Done Building Project "d:\repo\SH-Core-Libs\CLI\Options\Example2\CLI.Options.Example2.csproj" (default targets).
     5>C:\Program Files\dotnet\sdk\5.0.103\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(460,5): error NETSDK1029: Unable to use 'C:\Program Files\dotnet\packs\Microsoft.NETCore.App.Host.win-x64\5.0.3\runtimes\win-x64\native\apphost.exe' as application host executable as it does not contain the expected placeholder byte sequence '63-33-61-62-38-66-66-31-33-37-32-30-65-38-61-64-39-30-34-37-64-64-33-39-34-36-36-62-33-63-38-39-37-34-65-35-39-32-63-32-66-61-33-38-33-64-34-61-33-39-36-30-37-31-34-63-61-65-66-30-63-34-66-32' that would mark where the application name would be written. [d:\repo\SH-Core-Libs\CLI\Options\Example1\CLI.Options.Example1.csproj]
     5>Done Building Project "d:\repo\SH-Core-Libs\CLI\Options\Example1\CLI.Options.Example1.csproj" (default targets) -- FAILED.
     1>Done Building Project "d:\repo\SH-Core-Libs\SH-Core-Libs.sln" (default targets) -- FAILED.

-----

The VS build/test cycles seem to be working fine. It's only when I use dotnet build that I encounter this issue.

**Comment by jwdonahue** [comment]
These three test runs, demonstrate the issue on Az DevOps:

2 failures out of 20: https://dev.azure.com/sharperhacks/Core-Libs/_build/results?buildId=289&view=results
0 failures out of 20: https://dev.azure.com/sharperhacks/Core-Libs/_build/results?buildId=290&view=results
5 failures out of 20: https://dev.azure.com/sharperhacks/Core-Libs/_build/results?buildId=292&view=results

I've marked them all for retention. All were run against master 16744aa8920ba922d2fd235380546a6d3b7e8ac4.

**Comment by vitek-karas** [comment]
The weird part is that this is the only report of such problem we've seen. If the binary was somehow corrupted this would happen everywhere.

**Comment by jwdonahue** [comment]
I do seem to have a knack for breaking things, [but according to Google](https://www.google.com/search?q=%5Capphost.exe%27+as+application+host+executable+as+it+does+not+contain+the+expected+placeholder+byte+sequence+%2763-33-61-62-38-66-66-31-33-37-32-30-65-38-61-64-39-30-34-37-64-64-33-39-34-36-36-62-33-63-38-39-37-34-65-35-39-32-63-32-66-61-33-38-33-64-34-61-33-39-36-30-37-31-34-63-61-65-66-30-63-34-66-32%27+that+would+mark+where+the+application+name+would+be+written.), I am not the only one who has seen this in the past few months.

[This one](https://developercommunity.visualstudio.com/t/netsdk1029-error-strying-to-build-a-new-project-in/1190870) seemed to resolve itself with a VS reinstall, but that doesn't seem viable for my case.

My particular build hits this and another issue where apphost.exe is reported to be in use.  And I hit these on my work station as well as running builds on Az DevOps build agents, so it's probably my non-generic build configuration that somehow surfaces this issue.  I am sure ADO agents build lots of random stuff, but on my work-station, this is the only .NET project I've got active at the moment, and I only ever build one thing at a time.

**Comment by jwdonahue** [comment]
What if it's the code that is checking for the signature? Could it be that there's a hold-over value in a buffer somewhere, that causes the code to test the signature against a garbage oracle?

**Comment by jwdonahue** [comment]
@vitek-karas, do you have the access privileges in my project, that you need to diagnose this? I'll be happy to extend privileges to you, if you need them.

**Comment by jwdonahue** [comment]
The other related failure, looks like this:

C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018: The "CreateAppHost" task failed unexpectedly. [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018: System.IO.IOException: The process cannot access the file 'D:\a\1\s\.BuildOutput\net5.0\Obj\Debug\apphost.exe' because it is being used by another process. [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018:    at System.IO.FileSystem.CopyFile(String sourceFullPath, String destFullPath, Boolean overwrite) [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018:    at Microsoft.NET.HostModel.AppHost.BinaryUtils.CopyFile(String sourcePath, String destinationPath) [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018:    at Microsoft.NET.HostModel.AppHost.HostWriter.CreateAppHost(String appHostSourceFilePath, String appHostDestinationFilePath, String appBinaryFilePath, Boolean windowsGraphicalUserInterface, String assemblyToCopyResorcesFrom) [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018:    at Microsoft.NET.Build.Tasks.CreateAppHost.ExecuteCore() [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018:    at Microsoft.NET.Build.Tasks.TaskBase.Execute() [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]
C:\Program Files\dotnet\sdk\5.0.201\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(508,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [D:\a\1\s\CLI\Options\Example1\CLI.Options.Example1.csproj]

**Comment by jwdonahue** [comment]
And this is another seemingly unrelated random failure that only happens on the DevOps build:

D:\a\1\s\CLI\Options\Options\CLI.Options.csproj : error NU1101: Unable to find package JetBrains.Annotations. No packages exist with this id in source(s): Microsoft Visual Studio Offline Packages [D:\a\1\s\SH-Core-Libs.sln]

I've never seen this one when building locally.

**Comment by jwdonahue** [comment]
Found it! I fixed [my build bug](https://dev.azure.com/sharperhacks/Core-Libs/_workitems/edit/20). At some point I managed to have all of the projects sharing the same intermediate build path. I probably went a long time without hitting the problem. Perhaps adding one more project to the solution was just enough to tip it over? 

The other similar issues I could find online, all seemed to resolve to anti-virus or broken VS install. I verified my AV was not to blame and then noticed the path to the apphost.exe file did not have the project or assembly name embedded in it. A few iterations later, I've got that part fixed.  Still have the intermittent JetBrains.Annotations issue, but I'll start another bug for that if I can't resolve it on my own in the next day or two.

All the the apphost.exe issues are now behind me. Apologies for roping you folks into this.
